fixtures/get-platform-proxy-remote-bindings/tests/index.test.ts

```diff
@@ -15,123 +20,174 @@ const remoteWorkerName = `tmp-e2e-worker-test-remote-bindings-${randomUUID().spl
-				await env.MY_WORKER.fetch("http://example.com")
-			).text();
-			expect(workerText).toEqual(
-				"Hello from a remote Worker part of the getPlatformProxy remote bindings fixture!"
-			);
-
-			const kvValue = await env.MY_KV.get("test-key");
-			expect(kvValue).toEqual("remote-kv-value");
-
-			await dispose();
-		});
-
-		test("getPlatformProxy does not work with remote bindings if the experimental remoteBindings flag is not turned on", async () => {
-			const { env, dispose } = await getPlatformProxy<{
-				MY_WORKER: Fetcher;
-				MY_KV: KVNamespace;
-			}>({
-				configPath: "./.tmp/wrangler.json",
+			test("getPlatformProxy does not work with remote bindings if the experimental remoteBindings flag is not turned on", async () => {
+				const { env, dispose } = await getPlatformProxy<{
+					MY_WORKER: Fetcher;
+					MY_KV: KVNamespace;
+				}>({
+					configPath: "./.tmp/wrangler.json",
+				});
+
+				const response = await fetchFromWorker(
+					env.MY_WORKER,
+					"Service Unavailable"
+				);
+				const workerText = await response?.text();
+				expect(workerText).toEqual(
+					`[wrangler] Couldn\'t find \`wrangler dev\` session for service "${remoteWorkerName}" to proxy to`
+				);
+
+				const kvValue = await env.MY_KV.get("test-key");
+				expect(kvValue).toEqual(null);
+
+				await dispose();
 			});
-
-			const workerText = await (
-				await env.MY_WORKER.fetch("http://example.com")
-			).text();
-			expect(workerText).toEqual(
-				`[wrangler] Couldn\'t find \`wrangler dev\` session for service "${remoteWorkerName}" to proxy to`
-			);
-
-			const kvValue = await env.MY_KV.get("test-key");
-			expect(kvValue).toEqual(null);
-
-			await dispose();
-		});
-	});
+		},
+		{ timeout: 50_000 }
+	);
 } else {
 	test.skip("getPlatformProxy - remote bindings (no auth credentials)");
 }
 
+/**
+ * Tries to fetch from a worker multiple times until a response is returned which matches a specified
+ * statusText. Each fetch has a timeout signal making sure that it can't simply get stuck.
+ *
+ * This utility is used, instead of directly fetching from the Worker in order to prevent flakiness.
+ *
+ * @param worker The Worker to fetch from.
+ * @param expectedStatusText The response's expected statusText.
+ * @returns The successful Worker's response or null if no such response was obtained.
+ */
+async function fetchFromWorker(
+	worker: Fetcher,
+	expectedStatusText: string
+): Promise<Response | null> {
+	let response: Response | null = null;
+
+	// It seems like fetching from a remote worker might
+	// sometimes take quite a while...
+	const fetchTimeout = 5_000;
+
+	await vi.waitFor(
+		async () => {
+			try {
+				const maybeSuccessfulResponse = await worker.fetch(
+					"http://example.com",
+					{
+						signal: (global.AbortSignal as typeof AbortSignal).timeout(
+							fetchTimeout
+						),
+					}
+				);
+				if (maybeSuccessfulResponse.statusText === expectedStatusText) {
+					response = maybeSuccessfulResponse;
+				}
```

Comment:
```suggestion
				expect(maybeSuccessfulResponse.statusText).toEqual(expectedStatusText);
				return maybeSuccessfulResponse;
```

The return value from the callback will be returned from `waitFor()`.